utils/testserver/index.js

```diff
@@ -258,27 +263,31 @@ class TestServer {
     if (this._csp.has(pathName))
       response.setHeader('Content-Security-Policy', this._csp.get(pathName));
 
-    fs.readFile(filePath, (err, data) => {
-      if (err) {
-        response.statusCode = 404;
-        response.end(`File not found: ${filePath}`);
-        return;
-      }
-      const extension = filePath.substring(filePath.lastIndexOf('.') + 1);
-      const mimeType = extensionToMime[extension] || 'application/octet-stream';
-      const isTextEncoding = /^text\/|^application\/(javascript|json)/.test(mimeType);
-      const contentType = isTextEncoding ? `${mimeType}; charset=utf-8` : mimeType;
-      response.setHeader('Content-Type', contentType);
-      if (this._gzipRoutes.has(pathName)) {
-        response.setHeader('Content-Encoding', 'gzip');
-        const zlib = require('zlib');
-        zlib.gzip(data, (_, result) => {
-          response.end(result);
-        });
-      } else {
-        response.end(data);
-      }
-    });
+    const {err, data} = await readFileAsync(filePath).then(data => ({data})).catch(err => ({err}));
+    // The HTTP transaction might be already terminated after async hop here - do nothing in this case.
+    if (response.writableEnded)
+      return;
+    if (err) {
+      response.statusCode = 404;
+      response.end(`File not found: ${filePath}`);
+      return;
+    }
+    // after async hop, the request might've been already aborted, so attempting to change response
```

Comment:
Should this comment be placed next to the `writableEnded` check above?

Reply:
thanks! stray comment